rest/server_context_test.go

```diff
@@ -163,3 +165,197 @@ func (client *MockClient) RespondToGET(url string, response *http.Response) {
+	f()
+	return buf.String()
+}
+
+func TestStartReplicators(t *testing.T) {
+	defer base.SetUpTestLogging(base.LevelInfo, base.KeyReplicate)()
+	var replications []*ReplicationConfig
+
+	// Should be skipped; create_target option is not currently supported.
+	replications = append(replications, &ReplicationConfig{
+		Source:       "http://127.0.0.1:4985/imdb_us",
+		Target:       "http://127.0.0.1:4985/imdb_uk",
+		Async:        true,
+		CreateTarget: true})
+
+	// Should be skipped; doc_ids option is not currently supported
+	replications = append(replications, &ReplicationConfig{
+		Source: "http://127.0.0.1:4985/imdb_us",
+		Target: "http://127.0.0.1:4985/imdb_uk",
+		Async:  true,
+		DocIds: []string{"doc1", "doc2", "doc3", "doc4"}})
+
+	// Should be skipped; proxy option is not currently supported.
+	replications = append(replications, &ReplicationConfig{
+		Source: "http://127.0.0.1:4985/imdb_us",
+		Target: "http://127.0.0.1:4985/imdb_uk",
+		Async:  true,
+		Proxy:  "http://127.0.0.1:443/proxy"})
+
+	// Start a one-shot replication.
+	replications = append(replications, &ReplicationConfig{
+		Source:        "http://127.0.0.1:4985/imdb_us",
+		Target:        "http://127.0.0.1:4985/imdb_uk",
+		Async:         true,
+		ReplicationId: "58a632bb8d7e110445d3d65a98365d62"})
+
+	serverConfig := &ServerConfig{AdminInterface: &DefaultAdminInterface, Replications: replications}
+	serverContext := NewServerContext(serverConfig)
+	out := captureOutput(serverContext.startReplicators)
+
+	assert.Contains(t, "/_replicate create_target option is not currently supported", out)
+	assert.Contains(t, "/_replicate doc_ids option is not currently supported", out)
+	assert.Contains(t, "/_replicate proxy option is not currently supported", out)
+	assert.Contains(t, "ReplicationId:58a632bb8d7e110445d3d65a98365d62", out)
+}
+
+func TestPostStartup(t *testing.T) {
+	defer base.SetUpTestLogging(base.LevelInfo, base.KeyReplicate)()
+	var replications []*ReplicationConfig
+
+	// Start a one-shot replication.
+	replications = append(replications, &ReplicationConfig{
+		Source:        "http://127.0.0.1:4985/imdb_us",
+		Target:        "http://127.0.0.1:4985/imdb_uk",
+		Async:         true,
+		ReplicationId: "58a632bb8d7e110445d3d65a98365d62"})
+
+	serverConfig := &ServerConfig{AdminInterface: &DefaultAdminInterface, Replications: replications}
+	serverContext := NewServerContext(serverConfig)
+	out := captureOutput(serverContext.startReplicators)
+	assert.Contains(t, "ReplicationId:58a632bb8d7e110445d3d65a98365d62", out)
+}
+
+func TestGetOrAddDatabaseFromConfig(t *testing.T) {
+	//server := "walrus:"
+	//bucketName := "imbucket"
+
+	serverConfig := &ServerConfig{CORS: &CORSConfig{}, AdminInterface: &DefaultAdminInterface}
+	serverContext := NewServerContext(serverConfig)
+	//bucketConfig := BucketConfig{Server: &server, Bucket: &bucketName}
+
+	oldRevExpirySeconds := base.DefaultOldRevExpirySeconds
+	localDocExpirySecs := base.DefaultLocalDocExpirySecs
```

Comment:
Are these requred? I'd expect a nil value for these to automatically pull the defaults.

Reply:
Yeah, these values would set to default if not provided in the DbConfig. But the motivation of this test here is to cover the lines which use the value from DbConfig if provided; And used to give the default values while mocking up a fake DbConfig instead of custom values. I've updated this with custom values for better clarity about the intention of this test.